plugins/catalog-backend-module-ldap/src/ldap/config.ts

```diff
@@ -383,19 +421,41 @@ export function readProviderConfigs(config: Config): LdapProviderConfig[] {
       ? readTaskScheduleDefinitionFromConfig(c.getConfig('schedule'))
       : undefined;
 
+    const isUserList = Array.isArray(c.getOptional('users'));
+    const isGroupList = Array.isArray(c.getOptional('groups'));
+
     const newConfig = {
       id,
       target: trimEnd(c.getString('target'), '/'),
       tls: readTlsConfig(c.getOptionalConfig('tls')),
       bind: readBindConfig(c.getOptionalConfig('bind')),
-      users: readUserConfig(c.getConfig('users')),
-      groups: readGroupConfig(c.getConfig('groups')),
+      users: readUserConfig(
+        isUserList
+          ? c.getOptionalConfigArray('users')
+          : c.getOptionalConfig('users'),
+      ),
+      groups: readGroupConfig(
+        isGroupList
+          ? c.getOptionalConfigArray('groups')
+          : c.getOptionalConfig('groups'),
+      ),
       schedule,
     };
-    const merged = mergeWith({}, defaultConfig, newConfig, (_into, from) => {
-      // Replace arrays instead of merging, otherwise default behavior
-      return Array.isArray(from) ? from : undefined;
-    });
+
+    const merged = mergeWith(
+      {},
+      defaultConfig,
+      newConfig,
+      (_value, srcValue, key, object, _source) => {
+        // Remove users and groups from default if they are not set in the new config
+        if ((key === 'users' || key === 'groups') && !srcValue) {
+          return (object[key] = srcValue);
```

Comment:
This looks suspicious :) should probably break it up into more than one line just for clarity. Like the assignment and then return undefined or whatnot.

So what you are doing here is to literally remove the users and groups field, right? I'm not entirely sure why and if it does the right thing. What I can gather is, this code kinda compensates for the way that the `defaultConfig` looks, and the fact that we now happen to have those sometimes be array shaped. But I don't think that this is what you want to do. I think you want to instead go into `readUserConfig`, and _for every user entry_ (after normalization to an array as above), merge it with `defaultConfig.user`. Because each set of user settings should have these defaults applied to them; that's the way it worked before even though only one set of users was supported.

And then the same thing for groups.

Reply:
The mergeWith was already happening in existing code to avoid the arrays from being merged into an array. If the object specifies a value, the default value should not be added is what existing code meant. (eg. `options.attributes` from default should not be added to any specified option)

Now [this piece](https://github.com/lodash/lodash/issues/5242#issuecomment-1989313942) added that, if the EU did not mention a configuration for either groups or users, the default value should also not be merged as that would result in an object without `dn` or `filter`, but with a `map` and `set`. 
--> `undefined` users should not result in `map` and `set` only.

Reused but extracted existing callback as I went for a list approach as suggested. Now the `map` will decide to merge with default values or not (hence undefined will not merge, but arrays will also not copy).